src/mokernels/coregion.jl

```diff
@@ -0,0 +1,36 @@
+@doc raw"""
+    CoregionMOKernel(kernel::Kernel, B::AbstractMatrix)
+
+Kernel associated with the intrinsic coregionalization model.
+
+# Definition
+
+For inputs ``x, x'`` and output dimensions ``p_x, p_{x'}'``, the kernel is defined as[^ARL]
+```math
+k\big((x, p_x), (x, p_{x'})\big) = [B]_{p_x, p_{x'}} k^{x}\big(x, x'\big),
```

Comment:
IIRC we usually mention the parameters on the LHS:
```suggestion
k\big((x, p_x), (x', p_{x'}); B, \tilde{k}\big) = [B]_{p_x, p_{x'}} \tilde{k}\big(x, x'\big),
```